### ISSUE TITLE
Warning: Unknown event handler property

### ISSUE THREAD

**Issue by khrizt** [styled-components/styled-components#2218]
It looks like in some cases the props passed to a nested wrapped styled component end up in the final html tag when in v3 using extend did not.

I'm not sure if this is a bug or I'm using something incorrectly but as seeing that in the v3 this warning did not appear I chose to file a bug report instead of asking in stack overflow.

## System:
 - OS: Linux 4.15 Ubuntu 18.04.1 LTS (Bionic Beaver)
 - CPU: x64 Intel(R) Core(TM) i5-6500 CPU @ 3.20GHz
 - Memory: 483.48 MB / 15.53 GB
 - Container: Yes
 - Shell: 4.4.19 - /bin/bash
## Binaries:
 - Node: 9.10.0 - /usr/local/bin/node
 - Yarn: 1.12.3 - /usr/bin/yarn
 - npm: 5.6.0 - /usr/local/bin/npm
## npmPackages:
 - styled-components: ^4.0.0 => 4.1.1

## Reproduction

https://codesandbox.io/s/mq48yr83vy

## Steps to reproduce

Just access to this codesandbox at take a look at the console (I can't see it in the condesandbox console, but I can in the browser console).

## Expected Behavior

Not produce a warning.

## Actual Behavior

Produce a warning. In the v3 I had this as an extend and I never saw this warning.

**Comment by juanmagalhaes** [comment]
I'm having this issue too. I'm using a custom HOC that I created to enhance the component with a prop `onOutsideClick` and basically I pass all the props back to the wrapped component like this:

```
      render() {
        return (
          <div ref={this.setWrapperRef} style={wrapperStyle}>
            <WrappedComponent {...this.props} />
          </div>
        );
      }

```

I know that if I just filter this property out of `this.props` it will stop passing `onOutsideClick` to the wrapped component (Which can be a styled component) and the warning will go away, but it is annoying to have to do that. What if I really wanted to have a property called `onWhatever` would this warning be raised too?

**Comment by probablyup** [comment]
Thanks, this does seem to be a bug.

**Comment by mbrowne** [comment]
It seems to happen with any prop beginning with `on` regardless of case, e.g. we're seeing it with a prop named `onlyOneColumn`.

**Comment by mxstbr** [comment]
Do you reckon this is a bug in `is-valid-prop` @probablyup?

**Comment by probablyup** [comment]
Probably

**Comment by kokilaw** [comment]
Similar scenario like @juanmagalhaes stated. I also pass all the props to styled-component like this:
```
    render() {
        const { defaultChecked, disabled, id } = this.props
        return (
            <HorizontalWrapper {...this.props}>
                <input
                    id={id}
                    type="checkbox"
                    defaultChecked={defaultChecked}
                    disabled={disabled}
                    onChange={this.onChange}
                />
                <span />
            </HorizontalWrapper>
        )
    }
```
which includes `onColor`, `onColorShadow` and `onStateChange` where only  `onColor` and  `onColorShadow` are used in the styled-component. Anyway I am getting warned for all these three props. npm version is `4.1.2`.

**Comment by abumuawiyah** [comment]
I see the warning too (Styled-component - "styled-components": "^4.1.3")
This is because i passed all props ( {...other} ):

```jsx
<Container flexbox {...other}>
  <!-- codes -->
</Container>
```
```jsx
<Pagination 
      justifyContent="flex-end"
     onInit={(state) => console.log(state)}
     rowsPerPageOptions={[{id: 1, label: 10, value: 10},{id: 2, label: 20, value: 20}]}
     count={allItems.length}
     onChangePage={(currentPage) => console.log('currentPage', currentPage)}
     onChangeRowsPerPage={(rowsPerPage) => console.log('rowsPerPage', rowsPerPage)}
     controlProps={{label: 'Page'}}
/>
```
The reason why i passed all props is because, if the Styled is flex, I would be able to use flex properties like justifyContent etc

**Comment by coljung** [comment]
> It seems to happen with any prop beginning with on regardless of case, e.g. we're seeing it with a prop named onlyOneColumn.

Totally!

My function statement was called onTestClick. Once i renamed it to testclick, it worked.

**Comment by Anaizing** [comment]
Im currently in the middle of migrating from SC -v 2.2.1 to 4.2 and I have the same error in one of the projects... Is the only option to rename the handler?
![image](https://user-images.githubusercontent.com/29471810/58448922-c12d9b80-814c-11e9-85bf-ffb00bed80c7.png)

**Comment by mbrowne** [comment]
...or you can just ignore the warning. It won't show in production. You could also try the version 5 canary (`styled-components@5.0.0-4.canary-sheet`). I haven't tried it myself but I hear this will be fixed in version 5.

**Comment by probablyup** [comment]
We haven’t made any changes to prop filtering in v5 as of yet.

When wrapping a custom component with the styled() HOC, all props are
passed down. You’ll need to filter out DOM-irrelevant props yourself. We
only automatically filter props for primitive use cases like styled.div,
etc.

**Comment by mbrowne** [comment]
Thanks @probablyup. What is the likelihood at this point that more prop filtering (either as an opt-in feature or by default) will be added in version 5? Or does it depend on the availability of "toolchain support" as you mentioned at https://github.com/styled-components/styled-components/pull/2093#issuecomment-438144953?

**Comment by lak23shita** [comment]
I am facing with this warning and that's why it is ignoring all the firebase properties .How to fix this warning?
![Capture](https://user-images.githubusercontent.com/45456528/82744765-9d6e5700-9d9a-11ea-8002-551f70f52a6f.PNG)

**Comment by pmirandaarias** [comment]
This happen when some component has in it's interfaces a property like this:
`[key: string]: any;`
so then, you can put any `onWhatEver` props, and typescript won't alert you about nothing but it's not defined per se in the component, so take careful about that.

**Comment by saintpepsicola** [comment]
is there no fix for this yet?

**Comment by davidrochin** [comment]
No fix yet?

**Comment by gongruru** [comment]
No fix yet?
